### PR TITLE
Include commits since last release in RPM build number

### PR DESCRIPTION
**Pull request by zrhoffman**
This PR changes the build number to include the number of commits since the last release, rather than the total number of commits ever. Previously, the total commit count was used to ensure that a build from a later child commit will always be considered a higher version number.

But because the version number increases each time the release version increases, counting only the commits since the last release tag is sufficient.
<hr/>

## Which Traffic Control components are affected by this PR?

- Documentation
- CDN in a Box
- Traffic Portal v2

## What is the best way to verify this PR?

* Test `/about` TO API endpoint
* Make sure the *CDN-in-a-Box CI* GHA workflow picks up the already-built RPMs instead of rebuilding them all again.

## PR submission checklist
- [ ] This PR has tests 
- [x] This PR has documentation 
- [x] This PR has a CHANGELOG.md entry 
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

**Comment by ocket8888** [comment]
It looks like either the Makefile changes aren't matching up with the changes elsewhere for some reason, or possibly there's a place where it needs to be changed that's currently being missed.
```
cp: cannot stat '../../dist/trafficcontrol-cache-config-8.1.0-.5310b010.el8.x86_64.rpm': No such file or directory
```

Alternatively, RPM package archive files are totally capable of containing this information - does that _**really**_ need to be in the filename, too? I know I've said this before, but it seems very redundant to me.

**Comment by zrhoffman** [comment]
> It looks like either the Makefile changes aren't matching up with the changes elsewhere for some reason, or possibly there's a place where it needs to be changed that's currently being missed.
> 
> ```
> cp: cannot stat '../../dist/trafficcontrol-cache-config-8.1.0-.5310b010.el8.x86_64.rpm': No such file or directory
> ```

Fixed in f49cf08806 and 2262faf933

> 
> Alternatively, RPM package archive files are totally capable of containing this information - does that _**really**_ need to be in the filename, too?

Yes so that RPMs are sorted in directories, first by version, then by release number.

> I know I've said this before, but it seems very redundant to me.

Yes it is, but this PR isn't meant to change that. That's probably a bigger conversation that might get pushback from some people.

**Comment by ocket8888** [comment]
I checked the build artifacts for the checks on this branch - specifically Traffic Ops - and found that it produced

- `traffic_ops-8.1.0-0.d5e35540.el8.src.rpm`
- `traffic_ops-8.1.0-0.d5e35540.el8.x86_64.rpm`

so it looks like that's saying there have been zero commits since 8.1.0. That's incorrect not only because 8.1.0 doesn't technically exist (:P) but because this PR consists of a non-zero amount of commits.

tl;dr we got a number now, but it's not the right number.

**Comment by zrhoffman** [comment]
> so it looks like that's saying there have been zero commits since 8.1.0. That's incorrect not only because 8.1.0 doesn't technically exist (:P)

That's the version in [VERSION](https://github.com/apache/trafficcontrol/blob/14122bb913/VERSION)

> but because this PR consists of a non-zero amount of commits.
> 
> tl;dr we got a number now, but it's not the right number.

The GitHub Actions only clone the repo with a `fetch-depth` of 5 commits. Because no tags matching the pattern are found within the 5 commits, it defaults to `0`. So `0` is expected, in this case.

**Comment by ocket8888** [comment]
So was that always broken? Do we care if the build artifacts match what a user would build?

**Review by ocket8888 — APPROVED** [review]

**Review by rimashah25 — COMMENTED** [review]
The changes and explanation LGTM.
